xapian-applications/omega/postprocess.py

```diff
@@ -68,75 +67,82 @@ def generate_final_log(search_log, clicks_log):
-            did_to_count = {}
-
-            QUERYID, QUERY, HITS = 0, 1, 2
-            DOCID = 1
-
-            # Build map: qid_to_clicks = {qid: {did: click_count}}
-            for row2 in reader2:
-                qid, did = row2[QUERYID], row2[DOCID]
-
-                did_to_count = qid_to_clicks[qid]
-
-                # Check if did is already present in did_to_count
-                if did in did_to_count:
-                    # Update did_to_count[did]
-                    click_count = int(did_to_count[did])
-                    click_count += 1
-                    did_to_count[did] = str(click_count)
-                else:
-                    did_to_count[did] = '1'
-
-                qid_to_clicks[qid] = did_to_count
-
-            f.seek(0)
-
-            clicklist = []
-
-            for row in reader1:
-                # Skip rows with empty Query string
-                if row[QUERY] == '':
-                    continue
-
-                # Convert Hitlist from str to list
-                hits = row[HITS]
-                hits = hits.strip().split(',')
-                row[HITS] = hits
-
-                clicklist = hits[:]
-
-                # Update clicklist with click values stored in map.
-                if row[QUERYID] in qid_to_clicks:
-                    did_to_count = qid_to_clicks[row[QUERYID]]
-                    for index, did in enumerate(clicklist):
-                        if did in did_to_count:
-                            clicklist[index] = did + ':' + did_to_count[did]
-                        else:
-                            clicklist[index] = did + ':' + '0'
-
-                row.append(clicklist)
-                writer.writerow(row)
-    return
+    QUERYID, QUERY, HITS = 0, 1, 2
+    DOCID = 1
+
+    qid_to_clicks = collections.defaultdict(dict)
+    did_to_count = {}
+    clicklist = []
+
+    with open(clicks_log, 'r') as clicks_f:
+        clicks_reader = csv.reader(clicks_f, delimiter='\t')
+
+        # Build map: qid_to_clicks = {qid: {did: click_count}}
+        for row in clicks_reader:
+            qid, did = row[QUERYID], row[DOCID]
+
+            did_to_count = qid_to_clicks[qid]
+
+            # Check if did is already present in did_to_count
+            if did in did_to_count:
+                # Update did_to_count[did]
+                click_count = did_to_count[did]
+                click_count += 1
+                did_to_count[did] = click_count
+            else:
+                did_to_count[did] = 1
+
+            qid_to_clicks[qid] = did_to_count
+            print(qid_to_clicks[qid])
+
+    with open(search_log, 'r') as search_f, open(final_log, 'w+') as final_f:
+        search_reader = csv.reader(search_f, delimiter='\t')
+        writer = csv.writer(final_f, delimiter='\t')
+
+        # Add headers to final log file
+        writer.writerow(["QueryID", "Query", "Hits", "Offset", "Clicks"])
+
+        queries = []
```

Comment:
You should use a `set` for this, which will be much more efficient when checking if a query already exists in it.